.github/workflows/release.yml

```diff
@@ -0,0 +1,87 @@
+on:
+  push:
+    # We trigger this workflow on tag pushes that match v* (eg. `v1.2.3`)
+    tags:
+      - 'v*'
+
+name: Create Release
+jobs:
+  # This job name must match the `needs` field in `release` step
+  run-tests:
+    # This is mostly a copy/paste from test.yml
+    name: Tests
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Set up Go 1.13
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Set BRANCH_NAME
+      run: echo "::set-env name=GIT_BRANCH::${GITHUB_REF#refs/heads/}"
+
+    - name: Run Tests
+      run: go test -v -coverprofile=c.out -count=1 ./...
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+
+    needs: run-tests
```

Comment:
Question unrelated to this review: Why did we decide to use the `needs` keyword instead of `depends-on`?

Reply:
It's a GitHub action specific keyword: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds

Reply:
Ah interesting, I'll book a trip to the Github offices to have words with them